packages/beacon-node/src/chain/balancesCache.ts

```diff
@@ -35,7 +35,7 @@ export class CheckpointBalancesCache {
     const epochBoundaryRoot =
       epochBoundarySlot === state.slot ? blockRootHex : toHexString(getBlockRootAtSlot(state, epochBoundarySlot));
 
-    const index = this.items.findIndex((item) => item.epoch === epoch && item.rootHex == epochBoundaryRoot);
+    const index = this.items.findIndex((item) => item.epoch === epoch && item.rootHex === epochBoundaryRoot);
```

Comment:
both `item.rootHex` and `epochBoundaryRoot` are typed `string`. This diff is safe